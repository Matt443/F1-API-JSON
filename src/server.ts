import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}