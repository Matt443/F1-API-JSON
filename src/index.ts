import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
