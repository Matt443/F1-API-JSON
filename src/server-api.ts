import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
