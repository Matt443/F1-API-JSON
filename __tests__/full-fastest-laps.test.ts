import { getFullFastestLaps } from "../src/server";

test("fastest laps", async () => {
    expect(await getFullFastestLaps(2024)).toMatchSnapshot();
});
