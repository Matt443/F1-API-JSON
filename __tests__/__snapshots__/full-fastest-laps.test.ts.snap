// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fastest laps 1`] = `
[
  {
    "avgSpeed": "238.066",
    "code": "LEC",
    "lap": 56,
    "name": "Charles Leclerc",
    "number": 16,
    "position": "1",
    "team": "Ferrari",
    "time": "1:19.813",
    "timeOfDay": "16:20:30",
  },
  {
    "avgSpeed": "237.762",
    "code": "NOR",
    "lap": 49,
    "name": "Lando Norris",
    "number": 4,
    "position": "2",
    "team": "McLaren Mercedes",
    "time": "1:19.915",
    "timeOfDay": "16:11:12",
  },
  {
    "avgSpeed": "237.418",
    "code": "SAI",
    "lap": 48,
    "name": "Carlos Sainz",
    "number": 55,
    "position": "3",
    "team": "Ferrari",
    "time": "1:20.031",
    "timeOfDay": "16:09:42",
  },
  {
    "avgSpeed": "236.920",
    "code": "PIA",
    "lap": 54,
    "name": "Oscar Piastri",
    "number": 81,
    "position": "4",
    "team": "McLaren Mercedes",
    "time": "1:20.199",
    "timeOfDay": "16:18:04",
  },
  {
    "avgSpeed": "236.669",
    "code": "RUS",
    "lap": 53,
    "name": "George Russell",
    "number": 63,
    "position": "5",
    "team": "Mercedes",
    "time": "1:20.284",
    "timeOfDay": "16:17:14",
  },
  {
    "avgSpeed": "236.363",
    "code": "PER",
    "lap": 47,
    "name": "Sergio Perez",
    "number": 11,
    "position": "6",
    "team": "Red Bull Racing Honda RBPT",
    "time": "1:20.388",
    "timeOfDay": "16:08:52",
  },
  {
    "avgSpeed": "236.055",
    "code": "ALO",
    "lap": 52,
    "name": "Fernando Alonso",
    "number": 14,
    "position": "7",
    "team": "Aston Martin Aramco Mercedes",
    "time": "1:20.493",
    "timeOfDay": "16:15:52",
  },
  {
    "avgSpeed": "234.780",
    "code": "STR",
    "lap": 49,
    "name": "Lance Stroll",
    "number": 18,
    "position": "8",
    "team": "Aston Martin Aramco Mercedes",
    "time": "1:20.930",
    "timeOfDay": "16:12:01",
  },
  {
    "avgSpeed": "234.340",
    "code": "MAG",
    "lap": 44,
    "name": "Kevin Magnussen",
    "number": 20,
    "position": "9",
    "team": "Haas Ferrari",
    "time": "1:21.082",
    "timeOfDay": "16:05:23",
  },
  {
    "avgSpeed": "234.317",
    "code": "GAS",
    "lap": 51,
    "name": "Pierre Gasly",
    "number": 10,
    "position": "10",
    "team": "Alpine Renault",
    "time": "1:21.090",
    "timeOfDay": "16:15:20",
  },
  {
    "avgSpeed": "234.190",
    "code": "TSU",
    "lap": 46,
    "name": "Yuki Tsunoda",
    "number": 22,
    "position": "11",
    "team": "RB Honda RBPT",
    "time": "1:21.134",
    "timeOfDay": "16:08:01",
  },
  {
    "avgSpeed": "234.158",
    "code": "HUL",
    "lap": 46,
    "name": "Nico Hulkenberg",
    "number": 27,
    "position": "12",
    "team": "Haas Ferrari",
    "time": "1:21.145",
    "timeOfDay": "16:08:02",
  },
  {
    "avgSpeed": "233.887",
    "code": "RIC",
    "lap": 46,
    "name": "Daniel Ricciardo",
    "number": 3,
    "position": "13",
    "team": "RB Honda RBPT",
    "time": "1:21.239",
    "timeOfDay": "16:08:12",
  },
  {
    "avgSpeed": "233.634",
    "code": "ZHO",
    "lap": 49,
    "name": "Zhou Guanyu",
    "number": 24,
    "position": "14",
    "team": "Kick Sauber Ferrari",
    "time": "1:21.327",
    "timeOfDay": "16:12:44",
  },
  {
    "avgSpeed": "233.557",
    "code": "OCO",
    "lap": 45,
    "name": "Esteban Ocon",
    "number": 31,
    "position": "15",
    "team": "Alpine Renault",
    "time": "1:21.354",
    "timeOfDay": "16:07:18",
  },
  {
    "avgSpeed": "233.361",
    "code": "BOT",
    "lap": 48,
    "name": "Valtteri Bottas",
    "number": 77,
    "position": "16",
    "team": "Kick Sauber Ferrari",
    "time": "1:21.422",
    "timeOfDay": "16:11:19",
  },
  {
    "avgSpeed": "232.801",
    "code": "ALB",
    "lap": 46,
    "name": "Alexander Albon",
    "number": 23,
    "position": "17",
    "team": "Williams Mercedes",
    "time": "1:21.618",
    "timeOfDay": "16:08:08",
  },
  {
    "avgSpeed": "230.469",
    "code": "HAM",
    "lap": 11,
    "name": "Lewis Hamilton",
    "number": 44,
    "position": "18",
    "team": "Mercedes",
    "time": "1:22.444",
    "timeOfDay": "15:18:57",
  },
  {
    "avgSpeed": "228.608",
    "code": "VER",
    "lap": 3,
    "name": "Max Verstappen",
    "number": 1,
    "position": "19",
    "team": "Red Bull Racing Honda RBPT",
    "time": "1:23.115",
    "timeOfDay": "15:07:26",
  },
]
`;
